### PR TITLE
﻿Fix bugs in Keras BNFold and add unit tests

### PR DESCRIPTION
**Pull request by quic-geunlee**
Signed-off-by: Geunho Lee <quic_geunlee@quicinc.com>

### CODE REVIEW

File: TrainingExtensions/tensorflow/src/python/aimet_tensorflow/keras/batch_norm_fold.py [hunk 1]
```diff
@@ -313,8 +313,8 @@ def _get_weight_tensor_transpose_reshape(conv_linear: LAYER_TYPE) -> libpymo.Ten
     weight_tensor = libpymo.TensorParams()
 
     # linear array to be sent for bn fold
-    weight = conv_linear.kernel.numpy()
-    shape = conv_linear.kernel.shape
+    weight = conv_linear.get_weights()[0]
```
Comment (quic-akhobare) [inline TrainingExtensions/tensorflow/src/python/aimet_tensorflow/keras/batch_norm_fold.py:316]: What is 0 here? what does get_weights() return?

Comment (quic-geunlee) [reply TrainingExtensions/tensorflow/src/python/aimet_tensorflow/keras/batch_norm_fold.py:316]: Hi @quic-akhobare 
Return of [get_weights()](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#get_weights) depends on layer type.

If layer is Conv2D with bias, it will return `[weight_tensor, bias_tensor]`
else if layer is Conv2D without bias, it will return `[weight_tensor]`
else if layer is BatchNorm, it will return `[gamma, beta, moving_mean, moving_variance]`

I feel the name of get_weights is ambiguous, but it's implementation of TensorFlow :(

Comment (quic-akhobare) [reply TrainingExtensions/tensorflow/src/python/aimet_tensorflow/keras/batch_norm_fold.py:316]: Got it. PyTorch calls these weights too.. E.g. gamma maps to "weight" for PyTorch BN layers

**Review by quic-akhobare — APPROVED** [review]
